comixed-rest-api/src/main/java/org/comixedproject/messaging/ComiXedWebSocketChannelInterceptor.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * ComiXed - A digital comic book library management application.
+ * Copyright (C) 2021, The ComiXed Project
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses>
+ */
+
+package org.comixedproject.messaging;
+
+import java.util.List;
+import lombok.extern.log4j.Log4j2;
+import org.comixedproject.authentication.ComiXedUserDetailsService;
+import org.comixedproject.authentication.JwtTokenUtil;
+import org.comixedproject.messaging.model.StompPrincipal;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.simp.stomp.StompCommand;
+import org.springframework.messaging.simp.stomp.StompHeaderAccessor;
+import org.springframework.messaging.support.ChannelInterceptor;
+import org.springframework.messaging.support.MessageHeaderAccessor;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Component;
+
+/**
+ * <code>ComiXedWebSocketChannelInterceptor</code> provides a security interceptor to create the
+ * principal for the messaging layer.
+ *
+ * @author Darryl L. Pierce
+ */
+@Component
+@Log4j2
+public class ComiXedWebSocketChannelInterceptor implements ChannelInterceptor {
+  private static final String HEADER_STRING = "Authorization";
+
+  @Autowired private ComiXedUserDetailsService userDetailsService;
+  @Autowired private JwtTokenUtil jwtTokenUtil;
+
+  @Override
+  public Message<?> preSend(final Message<?> message, final MessageChannel channel) {
+    StompHeaderAccessor accessor =
+        MessageHeaderAccessor.getAccessor(message, StompHeaderAccessor.class);
+    if (accessor != null && StompCommand.CONNECT.equals(accessor.getCommand())) {
+      final List<String> authorization = accessor.getNativeHeader(HEADER_STRING);
+      if (authorization != null && !authorization.isEmpty()) {
```

Comment:
What do you think about using CollectionUtils.isEmpty from the spring framework or CollectionUtils.isNotEmpty from Apache Commons Collections both being Null-safe check ?

Stream and Optional can be used but I'm not sure whether it's clearer with something like : 
Optional<String> optionalToken = Optional.ofNullable(authorization).orElse(Collections.emptyList()).stream().flatMap(Stream::ofNullable).findFirst();
optionalToken.ifPresent(token -> {
        log.trace("Auth token found");
        doCreatePrincipal(accessor, token);
});

Reply:
I like it! I need to spend more time thinking in streams for things like this. 😄